From 9b92de58e54808402db8f70aac4777757c6d8ebc Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Mon, 30 Jul 2018 15:27:57 -0300
Subject: [PATCH 3/3] ostreemanager: tag /run/aktualizr/ostree-pending-update
 after updates

Have an easy way for userspace to know when there is a pending ostree
update, allowing custom automatic reboot mechanisms.

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
---
 src/libaktualizr/package_manager/ostreemanager.cc | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/libaktualizr/package_manager/ostreemanager.cc b/src/libaktualizr/package_manager/ostreemanager.cc
index 71562e13..04287e16 100644
--- a/src/libaktualizr/package_manager/ostreemanager.cc
+++ b/src/libaktualizr/package_manager/ostreemanager.cc
@@ -193,6 +193,16 @@ data::InstallOutcome OstreeManager::install(const Uptane::Target &target) const
   }
   LOG_INFO << "Performing sync()";
   sync();
+
+  LOG_INFO << "Creating /run/aktualizr/ostree-pending-update";
+  boost::filesystem::create_directory("/run/aktualizr");
+  std::ofstream file("/run/aktualizr/ostree-pending-update");
+  if (!file.good()) {
+    LOG_ERROR << "Failed to open /run/aktualizr/ostree-pending-update";
+  }
+  file.close();
+  LOG_INFO << "Created empty /run/aktualizr/ostree-pending-update";
+
   return data::InstallOutcome(data::UpdateResultCode::kOk, "Installation successful");
 }
 
-- 
2.18.0

